### PR TITLE
test: increase test coverage to 100%

### PR DESCRIPTION
**Pull request by nolanlawson**
It makes me nervous when the tests don't cover everything 100%.

- Sets `jest.config.js` to require a `coverageThreshold` of 100%
- Adds new tests to test existing behavior
- Adds `/* istanbul ignore */` comments where necessary
- Does not change any existing code behavior

### CODE REVIEW

**Comment by nolanlawson** [comment]
There are a few cases where I added `TODO`s next to the `istanbul ignore`s because it seemed impossible to me that the code would ever get called, but I didn't want to change existing behavior (for fear of introducing regressions).

Other cases like `process.env.NODE_ENV === 'production'` are pretty uncontroversial for adding `istanbul ignore`s.

File: test/reactive-handler.spec.ts [hunk 1]
```diff
@@ -815,6 +881,36 @@ describe('ReactiveHandler', () => {
...
+
+            property.foo.expando = 0;
+
+            expect([...property.foo]).toEqual(['bar', 'baz']);
+            expect(property.foo.expando).toEqual(0);
+
+            property.foo.expando = 0;
+
+            expect([...property.foo]).toEqual(['bar', 'baz']);
+            expect(property.foo.expando).toEqual(0);
```
Comment (jodarove) [inline test/reactive-handler.spec.ts:911]:
```suggestion
```

File: test/reactive-handler.spec.ts [hunk 2]
```diff
@@ -815,6 +881,36 @@ describe('ReactiveHandler', () => {
...
+            const obj = {foo: ['bar', 'baz'] };
+            const property = target.getProxy(obj);
+
+            property.foo.length = 0;
+
+            expect(property.foo).toEqual([]);
+
+            property.foo.length = 0;
+
+            expect(property.foo).toEqual([]);
```
Comment (jodarove) [inline test/reactive-handler.spec.ts:895]:
```suggestion
```

File: test/reactive-handler.spec.ts [hunk 3]
```diff
@@ -134,6 +134,32 @@ describe('ReactiveHandler', () => {
...
+        expect(value).toEqual('bar')
+        desc.set!('baz')
+        expect(value).toEqual('baz')
+    });
+    it('should understand undefined property desc', function () {
+        const target = new ReactiveMembrane();
+        const obj = {};
+        const property = target.getProxy(obj);
+        const desc = Object.getOwnPropertyDescriptor(property, 'fake');
+        expect(desc).toBe(undefined);
```
Comment (jodarove) [inline test/reactive-handler.spec.ts:161]:
```suggestion
        expect(desc).toBeUndefined();
```

File: test/reactive-dev-formatter.spec.ts [hunk 4]
```diff
@@ -13,16 +13,41 @@ describe('reactive dev formatter', function() {
...
+                    undefined
+                ]
+            }
+        ]);
+        expect(result[1].object[0].el).toBe(el);
+    });
+
+    it('body is always null', () => {
+        expect((window as any).devtoolsFormatters[0].hasBody({})).toBe(false);
+        expect((window as any).devtoolsFormatters[0].body({})).toBe(null);
```
Comment (jodarove) [inline test/reactive-dev-formatter.spec.ts:51]:
```suggestion
        expect((window as any).devtoolsFormatters[0].body({})).toBeNull();
```

**Review by jodarove — CHANGES_REQUESTED** [review]

File: test/reactive-handler.spec.ts (hunk 2 above)
Comment (nolanlawson) [reply test/reactive-handler.spec.ts:895]: This duplication is deliberate. I'll add a comment.

File: test/reactive-handler.spec.ts (hunk 1 above)
Comment (nolanlawson) [reply test/reactive-handler.spec.ts:911]: Ditto.

**Review by jodarove — APPROVED** [review]
lgtm

**Pull request merged**